roadmap/implementers-guide/src/protocol-disputes.md

```diff
@@ -0,0 +1,74 @@
+
+Now the dispute can be happening quite some time later than the inclusion, but also right during backing. As such the block is stored or more so it's solomon-reed encoded erasure chunks, from which the
+PoV can be reconstructed.
+
+A reconstructed PoV can be verified with the defined verification code, that is valid during the session the block was included or backed.
+If the block is invalid and there exists at least one backing vote and one validity challenging vote, a dispute exists.
+The existance of a dispute is detected by a backing checker
+or, if the block made it through backing stage, by an approval checker.
+In either case, the validator casts and distributes its vote via means of gossip.
+
+At this point the set of backing validators can not be trusted (since they voted for the block despite something being
+fishy at the very least). On the other hand, one must also consider, the validator that blew the whistle has ulterior motives
+to do so (i.e. it is controlled by a third party and wants to incur damage to itself).
+In either way, there are malicious validators around.
+As a consequence, all validators at the time of block backing, are being notified via broadcast of
+the first challenging vote.
+Validators that backed the candidate implicitly count as votes. Those validators are allowed to cast
+a regular vote (a non-backing vote) as well, but it is generally not in their interest to vote both sides, since that would
+advance the progress towards super majority either way and have their bonds slashed.
+If both votes lean in the same direction, i.e. both positive they are only counted as one.
+Two opposing votes by the same validator would be equal to an attempted double vote and would be slashed accordingly.
+
+All validators at block inclusion time are eligible to (and should) cast their Vote. The backing votes of backing checkers
+are counted as votes as well.
+
+## Initiation
+
+A dispute is initiated by one approval checker creating and gossiping a vote, that challenges the vote.
+approval checkers run the validation code and use this to verify the transactions.
+
+After an approval checker challenged a block, all validators that received the gossiped vote, reconstruct the block
+from availability erasure code chunks and check the block themselves via the validation code.
+The result of that check is converted into a vote, and distributed via the same mechanics as the first one.
+
+Once a receiver receives quorum and supermajority in one or the other direction, the
+vote is concluded.
+Conclusion implies that the result for this block can not be altered anymore, valid or invalid is fixed now.
+
+In order to ensure, the dispute result is not forgotten or intentionally side stepped, it has to be recorded on chain.
+This on chain recording mechanic must be vigilant, in a sense, that new emerging forks
+must also receive the dispute resolution recorded (transplantation) iff the disputed block is in the
```

Comment:
I agree with everything except for the 'iff' - we don't want malicious validators to be able to avoid slashing just because they reorganize the chain to no longer contain their attempt at including a bad parablock.